query-engine/driver-adapters/src/transaction.rs

```diff
@@ -6,10 +6,74 @@ use quaint::{
     Value,
 };
 
-use crate::proxy::{TransactionOptions, TransactionProxy};
+use crate::proxy::{TransactionContextProxy, TransactionOptions, TransactionProxy};
 use crate::{proxy::CommonProxy, queryable::JsBaseQueryable, send_future::UnsafeFuture};
 use crate::{JsObject, JsResult};
 
+pub(crate) struct JsTransactionContext {
+    tx_ctx_proxy: TransactionContextProxy,
+    inner: JsBaseQueryable,
+}
+
+// Wrapper around JS transaction context objects that implements Queryable. Can be used in place of quaint transaction,
+// context, but delegates most operations to JS
+impl JsTransactionContext {
+    pub(crate) fn new(inner: JsBaseQueryable, tx_ctx_proxy: TransactionContextProxy) -> Self {
+        Self { inner, tx_ctx_proxy }
+    }
+
+    pub async fn start_transaction(&self) -> quaint::Result<Box<JsTransaction>> {
+        self.tx_ctx_proxy.start_transaction().await
+    }
+}
+
+#[async_trait]
+impl Queryable for JsTransactionContext {
+    async fn query(&self, q: QuaintQuery<'_>) -> quaint::Result<ResultSet> {
+        self.inner.query(q).await
+    }
+
+    async fn query_raw(&self, sql: &str, params: &[Value<'_>]) -> quaint::Result<ResultSet> {
+        self.inner.query_raw(sql, params).await
+    }
+
+    async fn query_raw_typed(&self, sql: &str, params: &[Value<'_>]) -> quaint::Result<ResultSet> {
+        self.inner.query_raw_typed(sql, params).await
+    }
+
+    async fn execute(&self, q: QuaintQuery<'_>) -> quaint::Result<u64> {
+        self.inner.execute(q).await
+    }
+
+    async fn execute_raw(&self, sql: &str, params: &[Value<'_>]) -> quaint::Result<u64> {
+        self.inner.execute_raw(sql, params).await
+    }
+
+    async fn execute_raw_typed(&self, sql: &str, params: &[Value<'_>]) -> quaint::Result<u64> {
+        self.inner.execute_raw_typed(sql, params).await
+    }
+
+    async fn raw_cmd(&self, cmd: &str) -> quaint::Result<()> {
+        self.inner.raw_cmd(cmd).await
+    }
+
+    async fn version(&self) -> quaint::Result<Option<String>> {
+        self.inner.version().await
+    }
+
+    fn is_healthy(&self) -> bool {
+        self.inner.is_healthy()
+    }
+
+    async fn set_tx_isolation_level(&self, isolation_level: IsolationLevel) -> quaint::Result<()> {
+        self.inner.set_tx_isolation_level(isolation_level).await
+    }
+
+    fn requires_isolation_first(&self) -> bool {
+        self.inner.requires_isolation_first()
+    }
+}
```

Comment:
This is like the 3rd repetition of this pattern, we may introduce a macro at this point.